File: SCC5900_algoritmos/11_folhas/11_folhas.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n_books, budget;
    cin >> n_books >> budget;

    vector<int> prices(n_books);
    for (int i = 0; i < n_books; i++) {
        cin >> prices[i];
    }

    vector<int> pages(n_books);
    for (int i = 0; i < n_books; i++) {
        cin >> pages[i];
    }

    vector<vector<int>> dp(n_books + 1, vector<int>(budget + 1, 0));
    for (int i = 1; i <= n_books; i++) {
        for (int j = 0; j <= budget; j++) {
            if (prices[i - 1] > j) {
                dp[i][j] = dp[i - 1][j];
            } else {
                dp[i][j] = max(dp[i - 1][j], dp[i - 1][j - prices[i - 1]] + pages[i - 1]);
            }
        }
    }

    cout << dp[n_books][budget] << endl;

    return 0;
}

File: SCC5900_algoritmos/11_folhas/11_folhas.py

if __name__ == "__main__":
    n_books, budget = map(int, input().split())
    prices = list(map(int, input().split()))
    pages = list(map(int, input().split()))

    dp = [[0 for _ in range(budget + 1)] for _ in range(n_books + 1)]
    for i in range(1, n_books + 1):
        for j in range(budget + 1):
            if prices[i - 1] > j:
                dp[i][j] = dp[i - 1][j]
            else:
                dp[i][j] = max(dp[i - 1][j], dp[i - 1][j - prices[i - 1]] + pages[i - 1])
    print(dp[n_books][budget])